skia/RSkComponentRNSVGLinearGradient.cpp

```diff
@@ -0,0 +1,137 @@
+    gradientTransforMatrix_[SkMatrix::kMScaleX] = newRNSVGLinearGradientProps.gradientTransform[0];  //horizontal scale factor
+    gradientTransforMatrix_[SkMatrix::kMSkewY]  = newRNSVGLinearGradientProps.gradientTransform[1];  // vertical skew factor
+    gradientTransforMatrix_[SkMatrix::kMSkewX]  = newRNSVGLinearGradientProps.gradientTransform[2];  //horizontal skew factor
+    gradientTransforMatrix_[SkMatrix::kMScaleY] = newRNSVGLinearGradientProps.gradientTransform[3];  //vertical scale factor
+    gradientTransforMatrix_[SkMatrix::kMTransX] = newRNSVGLinearGradientProps.gradientTransform[4];  //horizontal translation
+    gradientTransforMatrix_[SkMatrix::kMTransY] = newRNSVGLinearGradientProps.gradientTransform[5];  // vertical translation
+  }
+
+  for (int i = 0; i < newRNSVGLinearGradientProps.gradient.size();) {
+
+  #ifdef ENABLE_RSKSVG_PROPS_DEBUG
+    RNS_LOG_INFO("gradient :: " <<i << " Item :: "<< newRNSVGLinearGradientProps.gradient[i]);
+    RNS_LOG_INFO("gradient :: " <<i+1 << " Item :: "<< (SharedColor)newRNSVGLinearGradientProps.gradient[i+1]);
+  #endif/*ENABLE_RSKSVG_PROPS_DEBUG*/
+
+    //The stop elements are ordered by offset followed by the color.
+    //StopColor received is a combined value with stop opacity. 
+    stopOffset_.push_back(newRNSVGLinearGradientProps.gradient[i]);
+    stopColor_.push_back(RSkColorFromSharedColor((SharedColor)newRNSVGLinearGradientProps.gradient[i+1], SK_ColorTRANSPARENT));
+    i=i+2;
+  }
+
+  setCommonNodeProps(newRNSVGLinearGradientProps);
+
+#ifdef ENABLE_RSKSVG_PROPS_DEBUG
+  for (auto item: stopOffset_) {
+    RNS_LOG_INFO("stopOffset :: " <<item);
+  }
+  for (auto item: stopColor_) {
+    RNS_LOG_INFO("stopColor_ ::\n"<<
+                 "R :: " << SkColorGetR(item)<<"\n"
+                 "G :: " << SkColorGetG(item)<<"\n"
+                 "B :: " << SkColorGetB(item)<<"\n"
+                 "A :: " <<SkColorGetA(item) <<"\n"
+    );
+  }
+  #endif /*ENABLE_RSKSVG_PROPS_DEBUG*/
+
+  invalidateLayer();
+
+  return RnsShell::LayerInvalidateAll;
+}
+
+void RSkComponentRNSVGLinearGradient::onSetAttribute(SkSVGAttribute attr, const SkSVGValue& attrValue) {
+
+  switch (attr) {
+    case SkSVGAttribute::kX1:
+      if (const auto* x1 = attrValue.as<SkSVGLengthValue>()) {
+        x1_ =*x1;
+      }
+      break;
+    case SkSVGAttribute::kY1:
+      if (const auto* y1 = attrValue.as<SkSVGLengthValue>()) {
+        y1_ =*y1;
+      }
+      break;
+    case SkSVGAttribute::kX2:
+      if (const auto* x2 = attrValue.as<SkSVGLengthValue>()) {
+        x2_ =*x2;
+      }
+      break;
+    case SkSVGAttribute::kY2:
+      if (const auto* y2 = attrValue.as<SkSVGLengthValue>()) {
+        y2_ =*y2;
+      }
+      break;
+    default:
+      this->INHERITED::onSetAttribute(attr, attrValue);
+  }
+}
+
+sk_sp<SkShader> RSkComponentRNSVGLinearGradient::getShader(const SkSVGRenderContext& ctx)  {
```

Comment:
Handle invalid brush negative case

Reply:
Right now, Invalid Brush Handled while calling get shader, by ensuring component is LinearGradient